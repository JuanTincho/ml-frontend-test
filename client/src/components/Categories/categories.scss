@import "../../styles/colors.scss";

.categories {
  color: $color-gray;
  margin-bottom: 16px;
  .last-category {
    font-weight: bold;
  }
}
