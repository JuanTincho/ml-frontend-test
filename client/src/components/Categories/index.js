import React from 'react';
import PropTypes from 'prop-types';
import './categories.scss';

const Categories = ({ categories }) => (
  <div className="categories">
    {categories.map((category, index) => {
      const lastCategory = index === categories.length - 1;

      return (
        <span key={category} className={lastCategory ? 'last-category' : null}>
          {!lastCategory ? `${category} > ` : category}
        </span>
      );
    })}
  </div>
);

Categories.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.string),
};

Categories.defaultProps = {
  categories: '',
};

export default Categories;
