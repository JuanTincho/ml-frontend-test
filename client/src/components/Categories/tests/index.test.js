import React from 'react';
import renderer from 'react-test-renderer';
import Categories from '../index';

describe('Categories', () => {
  const mockedCategories = ['category1', 'category2', 'category3'];

  it('renders correctly', () => {
    const tree = renderer.create(<Categories categories={mockedCategories} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
