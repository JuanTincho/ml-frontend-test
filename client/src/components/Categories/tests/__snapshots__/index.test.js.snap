// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders correctly 1`] = `
<div
  className="categories"
>
  <span
    className={null}
  >
    category1 &gt; 
  </span>
  <span
    className={null}
  >
    category2 &gt; 
  </span>
  <span
    className="last-category"
  >
    category3
  </span>
</div>
`;
